### PR TITLE
Add spelling corrections for resoruce

### PR DESCRIPTION
**Pull request by fishilico**
Many projects use "Resoruce" instead of "Resource": https://grep.app/search?q=resoruce

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -23128,6 +23128,8 @@ resopnses->responses
 resorce->resource
 resorces->resources
 resore->restore
+resoruce->resource
+resoruces->resources
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:23132]: Perhaps add resourced and resourcing equivalent typos too?

**Review by peternewman — CHANGES_REQUESTED** [review]
Just one comment

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (fishilico) [reply codespell_lib/data/dictionary.txt]: I agree and added these typos for all existing `->resource` and `->resources`.

**Comment by lurch** [comment]
I wonder if it'd be worth also adding the same typos but for 'source' (and variants) instead of 'resource'? :man_shrugging:

**Comment by fishilico** [comment]
> I wonder if it'd be worth also adding the same typos but for 'source' (and variants) instead of 'resource'?

The dictionary already contains corrections for `soruce` and `soruces`, so there is no need to add them: https://github.com/codespell-project/codespell/blob/89be12128e05a3bcc98194c7115991ac975ea1e4/codespell_lib/data/dictionary.txt#L24781-L24782

But the dictionary does not contain `soruced` nor `sorucing` and some open-source projects did these misspellings: https://grep.app/search?q=soruced and https://grep.app/search?q=sorucing. Would you prefer that I add these corrections to this PR, or that I open a new PR about "source and its derived words"?

**Comment by lurch** [comment]
> Would you prefer that I add these corrections to this PR, or that I open a new PR about "source and its derived words"?

That's a question for @peternewman ...

**Comment by peternewman** [comment]
> > Would you prefer that I add these corrections to this PR, or that I open a new PR about "source and its derived words"?
> 
> That's a question for @peternewman ...

Sorry for the delay. A separate PR is probably best as this one is already relatively long and those new words may bring up other alternative suggestions that want looking at.

**Pull request merged**